packages/core/src/builder.ts

```diff
@@ -440,6 +438,13 @@ export default class SchemaBuilder<Types extends SchemaTypes> {
 
     this.configStore.addTypeConfig(config, ref);
 
+    this.configStore.onTypeConfig(ref, () => {
+      const unionTypes = typeof types === 'function' ? types() : types;
+      unionTypes.forEach((type) => {
+        verifyRef(type);
```

Comment:
I don't think that this will be necessary, as mentioned above, this is mostly to detect when refs are undefined which I've only seen happen because of import issues.  The lazy evaluation should help with that